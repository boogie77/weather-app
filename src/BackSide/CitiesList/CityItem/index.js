import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

export default class CityItem extends Component {
  static propTypes = {
    onSelect: PropTypes.func,
    isSelected: PropTypes.bool,
    city: PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  };

  onClick = () => {
    const { onSelect, city } = this.props;
    onSelect(city);
  };

  render() {
    const { city, isSelected } = this.props;
    return (
      <li
        className={classNames("list-item", { "is-selected": isSelected })}
        onClick={this.onClick}
      >
        {city.title}
      </li>
    );
  }
}
