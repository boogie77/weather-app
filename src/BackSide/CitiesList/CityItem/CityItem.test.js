import React from "react";
import CityItem from "./index";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

const mockCity = {
  title: "Mock City"
};
const mockOnClick = jest.fn();

describe("<CityItem />", () => {
  describe("when is selected", () => {
    it("renders and matches the snapshot", () => {
      const wrapper = shallow(<CityItem isSelected city={mockCity} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("when is not selected", () => {
    it("renders and matches the snapshot", () => {
      const wrapper = shallow(<CityItem city={mockCity} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  it("calls the provided onSelect callback when clicked ", () => {
    const wrapper = shallow(
      <CityItem onSelect={mockOnClick} city={mockCity} />
    );
    wrapper.simulate("click");
    expect(mockOnClick).toHaveBeenCalled();
  });
});
