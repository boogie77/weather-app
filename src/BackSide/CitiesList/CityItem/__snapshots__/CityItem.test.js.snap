// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityItem /> when is not selected renders and matches the snapshot 1`] = `
<li
  className="list-item"
  onClick={[Function]}
>
  Mock City
</li>
`;

exports[`<CityItem /> when is selected renders and matches the snapshot 1`] = `
<li
  className="list-item is-selected"
  onClick={[Function]}
>
  Mock City
</li>
`;
